packages/xchain-ethereum/src/utils.ts

```diff
@@ -353,3 +354,31 @@ export const getDecimal = async (asset: Asset, provider: providers.Provider): Pr
     throw new Error(`Invalid provider: ${err}`)
   }
 }
+
+/**
+ * Get Token Balances
+ *
+ * @param {Array<TokenBalance>} tokenBalances
+ * @returns {Array<Balance>} the parsed balances
+ *
+ */
+export const getTokenBalances = (tokenBalances: TokenBalance[]): Balances => {
+  return tokenBalances.reduce((acc, cur) => {
+    const decimals = parseInt(cur.tokenInfo.decimals)
```

Comment:
Maybe `cur.tokenInfo.decimals` -> `cur?.tokenInfo?.decimals ?? NaN` (just in case `decimals` won't have a value, but we do need)

And later in line `369`:
```
if (validateSymbol(symbol) && validateAddress(tokenAddress) && decimals) { ... }
```